BabbleApp/babbleapp.py

```diff
@@ -80,9 +81,17 @@ async def check_for_updates(config, notification_manager):
     if config.settings.gui_update_check:
         try:
             response = requests.get(
-                "https://api.github.com/repos/Project-Babble/ProjectBabble/releases/latest"
+                "https://api.github.com/repos/Project-Babble/ProjectBabble/releases/latest",
+                timeout=10  # Add timeout
             )
-            latestversion = response.json()["name"]
+            response.raise_for_status()  # Will raise exception for HTTP errors
+            
+            data = response.json()
+            latestversion = data.get("name")
+            
+            if not latestversion:
+                print(f'[{lang._instance.get_string("log.warn")}] {lang._instance.get_string("babble.invalidVersionFormat")}')
```

Comment:
There is no locale string for `babble.invalidVersionFormat`. This will need to be added per locale file, if you do not feel comfortable translating per locale just leave the string as English.